evalml/tests/component_tests/test_log_transformer.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import pandas as pd
+import pytest
+import woodwork as ww
+
+from evalml.pipelines.components import LogTransformer
+
+
+def test_log_transformer_init():
+    log_ = LogTransformer()
+    assert log_.parameters == {}
+
+
+def test_log_transformer_no_y(X_y_regression):
+    X, y = X_y_regression
+    y = None
+
+    output_X, output_y = LogTransformer().fit_transform(X, y)
+    np.testing.assert_equal(X, output_X)
+    assert not output_y
+
+
+@pytest.mark.parametrize("input_type", ["np", "pd", "ww"])
+@pytest.mark.parametrize("data_type", ["positive", "mixed", "negative"])
+def test_log_transformer_fit_transform(data_type, input_type, X_y_regression):
+    X_, y_ = X_y_regression
+    if data_type == "positive":
+        y_ = np.abs(y_)
+    elif data_type == "negative":
+        y_ = -np.abs(y_)
+
+    if y_.min() <= 0:
+        y_ = y_ + abs(y_.min()) + 1
+    expected_log = np.log(y_)
+
+    X_ = pd.DataFrame(X_)
+    y_ = pd.Series(y_)
+
+    X, y = X_, y_
```

Comment:
I think you should copy X and y earlier in this test. Right now, if the `data_type` is mixed or negative, you change `y_` to become all positive (line 33), and when you set `y = y_`, it sets `y` to the newly-changed value of `y_`.

I think you should copy `X, y` after line 30, where you've altered the `y_` data to be in sync with the `data_type` but haven't yet made the values all positive.

Reply:
Good point!